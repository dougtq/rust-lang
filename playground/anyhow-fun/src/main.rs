use anyhow::anyhow;
use std::error::Error as StdError;

fn main() {
    println!("Hello, world!i AnyHow!!! ");

    let error = Box::<dyn StdError + Send + Sync>::from("oh no!");
    assert_eq!("oh no!", error.to_string());

    let error = anyhow!(error);
    assert_eq!("oh no!",
            error
             .downcast_ref::<Box<dyn StdError + Send + Sync>>()
             .unwrap()
             .to_string()
    );

    let error2 = anyhow!("oh no!").context("it failed");
    assert_eq!("oh no!", error2.source().unwrap().to_string());
}
