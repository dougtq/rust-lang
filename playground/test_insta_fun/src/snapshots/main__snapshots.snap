---
source: src/main.rs
expression: value
---
[
    1,
    2,
    3,
]
